nixos/modules/services/misc/mediatomb.nix

```diff
@@ -256,31 +326,33 @@ in {
 
   ###### implementation
 
-  config = mkIf cfg.enable {
+  config = let binaryCommand = "${pkg}/bin/${name}";
+               interfaceFlag = optionalString ( cfg.interface != "") "--interface ${cfg.interface}";
+               configFlag = optionalString (! cfg.customCfg) "--config ${pkgs.writeText "config.xml" configText}";
+    in mkIf cfg.enable {
     systemd.services.mediatomb = {
-      description = "MediaTomb media Server";
+      description = "${cfg.serverName} media Server";
       after = [ "network.target" ];
       wantedBy = [ "multi-user.target" ];
-      path = [ pkgs.mediatomb ];
-      serviceConfig.ExecStart = "${pkgs.mediatomb}/bin/mediatomb -p ${toString cfg.port} ${if cfg.interface!="" then "-e ${cfg.interface}" else ""} ${if cfg.customCfg then "" else "-c ${mtConf}"} -m ${cfg.dataDir}";
+      serviceConfig.ExecStart = "${binaryCommand} --port ${toString cfg.port} ${interfaceFlag} ${configFlag} --home ${cfg.dataDir}";
       serviceConfig.User = "${cfg.user}";
     };
 
-    users.groups = optionalAttrs (cfg.group == "mediatomb") {
-      mediatomb.gid = gid;
+    users.groups = {
+      "${cfg.group}".gid = gid;
     };
 
-    users.users = optionalAttrs (cfg.user == "mediatomb") {
-      mediatomb = {
+    users.users = {
+      "${cfg.user}" = {
         isSystemUser = true;
         group = cfg.group;
         home = "${cfg.dataDir}";
         createHome = true;
-        description = "Mediatomb DLNA Server User";
+        description = "${name} DLNA Server User";
       };
     };
 
-    networking.firewall = {
+    networking.firewall.interfaces."${cfg.interface}" = {
```

Comment:
I'm a bit uneasy with that one by the way.
It's coming from the commit I started upon.

I don't really know how to deal with the case where the `cfg.interface` is not specified (which is the default case, as empty string...).

Do you have any hints here?

Thanks in advance.

Reply:
Also I read we should not deal with the firewall rules within the service.
I kept this though as it's already the existing behavior...
(And the PR is getting a "tad" big already... ;)

Reply:
Do a quick `grep` for `openFirewall`. For this service I would suggest adding an `openFirewall` option and then do something like this (maybe a bit cleaner, though :smile:):
```
networking.firewall = mkMerge [
  (mkIf (cfg.openFirewall && cfg.interface != "") {
    interfaces."${cfg.interface}" = {
      allowedUDPPorts = [ 1900 cfg.port ];
      allowedTCPPorts = [ cfg.port ];
    };
  })
  (mkIf (cfg.openFirewall && cfg.interface == "") {
      allowedUDPPorts = [ 1900 cfg.port ];
      allowedTCPPorts = [ cfg.port ];
  })
];
```

Reply:
Oh yeah, thanks!
That's clearer (pushed a new commit with this, I removed the set duplication but that's about it)